src/components/LocationInput.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+import * as React from 'react';
+import {StyleSheet, Text, View} from 'react-native';
+import StringInput from './StringInput';
+import type {User} from '../types.js';
+
+const styles = StyleSheet.create({
+  label: {
+    fontSize: 18,
+    marginVertical: 10,
+  },
+  container: {
+    marginLeft: 10,
+  },
+});
+
+const LocationInput = ({
+  label,
+  user,
```

Comment:
It's about Component Design. LocationInput don't need to know `user` object. It only needs to know `location` object. and function `writeValue` should be function `onChangeLocation`

Reply:
Thank you for your feedback. I have rewritten the component accordingly.